languages/ruby/ruby_on_rails.md

```diff
@@ -101,25 +125,25 @@ This is a fancy way of saying that your views and controller code is not the pla
 ## Smells to watch for:
 Controller actions that set attributes on a variety of objects, i.e. your controller should not understand how one model relates to another and how they should be orchestrated.
 Controller actions that are longer than 10 lines of code (there’s no magic number here, but consider that when you see long actions.
+
 7 Patterns to Refactor Fat ActiveRecord Models include:
-extract value objects (and place in app/models)
-extract service objects (and place in app/services)
-extract form objects (and place in app/forms)
-extract query objects (and place in app/models)
-introduce view objects
-extract policy objects
-extract decorators (consider using the draper gem)
 
-# Security
+- extract value objects (and place in app/models)
+- extract service objects (and place in app/services)
+- extract form objects (and place in app/forms)
+- extract query objects (and place in app/models)
+- introduce view objects (helpers, ViewComponents and Decorators are all good directions to think in)
+- extract decorators (consider using the draper gem)
+- extract policy objects (there are a number of good gems for this)
 
-As a starting point, please read http://guides.rubyonrails.org/security.html and become familiar with the concepts.  We must adhere to all of the points mentioned there.
+# Security
 
-Another useful, quick read is http://blog.codeclimate.com/blog/2013/03/27/rails-insecure-defaults/
```

Comment:
I had removed this as it was a 404 - but I've found a new link for it.  That new link has been updated but its maybe less useful now than it used to be.  https://codeclimate.com/blog/rails-insecure-defaults

Who thinks we should include it? ✋

Reply:
I'm not in favor of a URL with 2013 in it :P maybe there's a newer/fresher one out there, or prob best to avoid as many external URLs as possible.

Reply:
Yes, the one I posted above is the updated version (looks like they re-did all their blog URLs to remove the dates).

I think its likely the same article, but it was updated in March.  That being said.... I'm not totally sure its as relevant today as it was back when this was originally added.  They've marked most of their original points as fixed and there aren't as many remaining.  Of those I think that "yes" we should be aware of, but some of them are a stretch and they're less "by default" now.  

I'd argue at this point that a rails setup has more and better built-in security than many of the other options out there (e.g. express) and that in my opinion is one of the selling points.  That's not to say you don't have to be smart and know the framework though!